### DISCUSSION TITLE
Incorrect filling of a polygon with a hole.

### DISCUSSION THREAD

**Discussion by jonimnim** [mapsforge/mapsforge#1287] [Q&A] [answered]
Hello. Again there was a problem with incorrect filling, but of a slightly different plan. Now it is pouring what should not be poured. Those. there is a polygon with a hole, but for some reason the hole at the polygon is also flooded. Please help me figure it out.
In the file I attach osm files and a map file that turned out, as well as a theme and a file with tags, as well as pictures of what should be and what happened
[mapsforge_island.zip](https://github.com/mapsforge/mapsforge/files/7987480/mapsforge_island.zip)
![island_true](https://user-images.githubusercontent.com/16680576/152177280-c89fcf11-3b36-4c90-8533-4adc5b032267.png)
![island_false](https://user-images.githubusercontent.com/16680576/152177300-a2e4fcde-7234-46cb-a448-d4000c8897c5.png)
.

**Comment by devemux86** [comment, marked as answer]
If you mean maps with coastlines, see the [map creation guide](https://github.com/mapsforge/mapsforge/blob/master/docs/MapCreation.md) about how to include sea + land + map data.
It describes the separate osm files, the tag-mapping and the render theme.
